doc/dev/adr/0020-large-message-chunking.md

```diff
@@ -0,0 +1,100 @@
+﻿# ADR 20: Large Message Chunking in MQTT Protocol
+
+## Status
+Proposed
+
+## Context
+The MQTT protocol has inherent message size limitations imposed by brokers and network constraints. Azure IoT Operations scenarios often require transmitting payloads that exceed these limits (e.g., firmware updates, large telemetry batches, complex configurations). Without a standardized chunking mechanism, applications must implement their own fragmentation strategies, leading to inconsistent implementations and interoperability issues.
+
+## Decision
+We will implement sdk-level message chunking as part of the Protocol layer to transparently handle messages exceeding the MQTT broker's maximum packet size.
+
+**The chunking mechanism will**:
+- Be enabled/disabled by a configuration setting.
+- Use standardized user properties for chunk metadata:
+   - The `__chunk` user property will contain a JSON object with chunking metadata.
+   - The JSON structure will include:
```

Comment:
I remember getting some pushback from @vaavva about introducing a JSON library dependency into our protocol layer. 

To avoid that, it may be easier to just set these 5 fields as 5 MQTT user properties rather than putting them all in one MQTT user property

Reply:
To clarify, the pushback I mentioned was about an orthogonal change (error modeling in headers).

I'm fine with using JSON here because System.Text.Json isn't a hard dependency for us to take, but double check with Val that the rust folks are on board. If they aren't, we may need to use some other serialization method here.

Reply:
This would all just end up being internally used, right? If so, I don't care - we can just use the JSON library internally. I would only have hesitation if it crossed API boundary, which I can't see why it would.

However, even then, given the pre-defined structure, we could just deserialize it to a struct that mirrors the JSON 1:1.